components/modals/deleteOperationModal.tsx

```diff
@@ -0,0 +1,108 @@
+import IconX from "tabler-icons/tsx/x.tsx";
+import { Audience } from "snitch-protos/protos/sp_common.ts";
+import { Pipeline } from "snitch-protos/protos/sp_pipeline.ts";
+import { getAudienceOpRoute } from "../../lib/utils.ts";
+import { opModal } from "../serviceMap/opModalSignal.ts";
+import { opUpdateSignal } from "../../islands/serviceMap.tsx";
+import { Toast, toastSignal } from "../toasts/toast.tsx";
+import { useState } from "preact/hooks";
+
+export const DeleteOperationModal = (
+  { audience, pipeline }: {
+    audience: Audience;
+    pipeline?: Pipeline;
+  },
+) => {
+  const [open, setOpen] = useState(true);
+  const close = () => opModal.value = null;
+
+  const deleteOp = async () => {
+    const response = await fetch(
+      `${getAudienceOpRoute(audience)}/delete`,
+      {
+        method: "POST",
+      },
+    );
+
+    const { success } = await response.json();
+
+    if (success.message) {
+      toastSignal.value = {
+        id: "operationDelete",
+        type: success.status ? "success" : "error",
+        message: success.message,
+      };
+      opModal.value.delete = false;
+      setOpen(false);
+      opUpdateSignal.value = null;
+    }
+    setTimeout(() => {
```

Comment:
Do you need to wait 3 seconds to close this? Better not to if it's not necessary.

Reply:
Removed this!